fix: editorials and terminology used (#12)
peppelinux/draft-demarco-acme-openid-federation: draft-demarco-acme-openid-federation.md

   protocol for OpenID Federation 1.0. If not, the Requestor starts the
   discovery process to find which are the Issuers within the federation.

2. The Requestor requests and obtains a new nonce from the Certificate Issuer,
   by sending a HTTP HEAD request to the Issuer's `newNonce` resource;

3. The Certificate Issuer evaluates the trust to the Requestor by checking if it is part of
   the federation. If not the `newNonce` request MUST be rejected (**TBD** the
   error to return). There are two ways the Certificate Issuer is able to check if a
   Requestor is part of the federation, these are listed below:

>     - The Requestor adds the Trust Chain JWT header parameter related to itself,
>       this option is RECOMMENDED since it reduces the effort of the Certificate Issuer in

Reviewer: ```suggestion
    - The Requestor adds the Trust Chain JWT header parameter related to itself.
      This option is RECOMMENDED since it reduces the effort of the Certificate Issuer in
```
Editorial

      evaluating the trust to the Requestor;

>     - The Requestor doesn't add the Trust Chain in the request, then the Certificate Issuer
>       MUST start a [Federation Entity
>       Discovery](https://openid.net/specs/openid-federation-1_0.html#section-8)
>       to obtain the Trust Chain related to the Requestor.

Reviewer: ```suggestion
    - The Requestor doesn't add the Trust Chain in the request. The Certificate Issuer
      MUST start a [Federation Entity
      Discovery](https://openid.net/specs/openid-federation-1_0.html#section-8)
      to obtain the Trust Chain related to the Requestor.
```
Similar to above, reads a little bit easier.

7.1.1](https://datatracker.ietf.org/doc/html/rfc8555#section-7.1.1) of [RFC8555]
in the federation Entity Configuration of the Issuer.

> TBD: the example below is non-normative. A table describing each REQUIRED or OPTIONAL

Author: ```suggestion
Below is a non-normative example about the Credential Issuer metadata object. All the parameters contained within it are defined in the [RFC8555].
```

   }
~~~~

The Requestor responds with an object with the following format:

sig (required, string):  a base64url encoding of a JWT, signing the token
    encoded in UTF-8 with one of the keys published in the requestor's OpenID
    Federation Entity Configuration: either in the top-level `jwks` claim as
    defined in Section 3 of [OIDC-FED] or referenced by the `signed_jwks_uri`,
    `jwks_uri`, or `jwks` claims in the entity metadata as defined in Section
    5.2.1 of [OIDC-FED]. It is REQUIRED that this JWT include a `kid` claim
    corresponding to a valid key; The Credential Issuer MUST only use keys with a
    corresponding `kid` value when evaluating the challenge response. Otherwise,
    the Certificate Issuer SHOULD enumerate all valid keys and accept a

>     signature from any of them.

Reviewer: I think this sentence is unnecessary? The certificate issuer shouldn't have to enumerate all keys, since in this changeset the requestor is REQUIRED to include valid key ID.

trust_chain (optional, array of string):  an array of base64url-encoded bytes

>     containing a signed JWT and representing the Trust Chain about the Requestor.

Reviewer: ```suggestion
    containing a signed JWT and representing the Trust Chain of the Requestor.
```
editorial

   }
~~~~

The Requestor responds with an object with the following format:

sig (required, string):  a base64url encoding of a JWT, signing the token
    encoded in UTF-8 with one of the keys published in the requestor's OpenID
    Federation Entity Configuration: either in the top-level `jwks` claim as
    defined in Section 3 of [OIDC-FED] or referenced by the `signed_jwks_uri`,
    `jwks_uri`, or `jwks` claims in the entity metadata as defined in Section
    5.2.1 of [OIDC-FED]. It is REQUIRED that this JWT include a `kid` claim
    corresponding to a valid key; The Credential Issuer MUST only use keys with a
    corresponding `kid` value when evaluating the challenge response. Otherwise,

>     the Certificate Issuer SHOULD enumerate all valid keys and accept a
>     signature from any of them.

Author: ```suggestion
```

the given entity (possibly performing Federation Entity Discovery to do so),
then:

* Verifies that the requested domain names match the FQDN contained within the
  `sub` parameter of the requestor's Entity Configuration. For example, if the
  `sub` parameter within the Entity Configuration contains the value
  `https://requestor.example.com/oidc/rp`, the extracted FQDN is then
  `requestor.example.com`. Since the Entity Configuration MUST contain at most
  one FQDN, this effectively means that this challenge type works with requests
  for a single domain name only.

* Verifies that the `sig` field of the payload includes a valid JWT over the
  challenge token, signed with one of the keys published in the Requestor's
  Entity Configuration, either in the top-level `jwks` claim as defined in
  Section 3 of [OPENID-FED] or referenced by the `signed_jwks_uri`, `jwks_uri`, or
  `jwks` claims in the entity metadata as defined in Section 5.2.1 of
  [OPENID-FED]. If the Requestor provides a `kid` value in its challenge response,
  only keys (JWKs) in the Entity Configuration with a matching `kid` value are

>   considered.

Reviewer: As earlier, since the requestor is now required to provide the `kid`, this can be reworded.

Maybe something like "Only keys in the entity configuration matching the `kid` in the requestor's challenge response are considered."

   }
~~~~

On receiving a response, the Certificate Issuer retrieves the public keys associated with
the given entity (possibly performing Federation Entity Discovery to do so),
then:

* Verifies that the requested `openid-federation` value matches the `sub`
  parameter of the Requestor's Entity Configuration. Since the Entity
  Configuration MUST contain at most one Entity Identifier, this effectively
  means this challenge type works with requests for a single Federation Entity
  only.

* Verifies that the `sig` field of the payload includes a valid JWT over the
  challenge token, signed with one of the keys published in the Requestor's
  Entity Configuration, either in the top-level `jwks` claim as defined in
  Section 3 of [OPENID-FED] or referenced by the `signed_jwks_uri`, `jwks_uri`, or
  `jwks` claims in the entity metadata as defined in Section 5.2.1 of

>   [OPENID-FED]. If the Requestor provides a `kid` value in its challenge response,
>   only keys (JWKs) in the Entity Configuration with a matching `kid` value are

Author: ```suggestion
  [OPENID-FED]. Since the Requestor MUST provide a `kid` value in its challenge response,
  only keys (JWKs) in the Entity Configuration with a matching `kid` value are
```

# Publication of the Certificates within the Federation

**TBD**, when the Certificate Issuer is the Trust Anchor or Intermediate, the X.509
Certificate linked to Federation Entity Key represented in JWK in the Subordinate
Statement related to the Requestor, SHOULD be extended with the claim `x5c`,
containing the issued X.509 Certificate.

# Certificate Lifecycle and Revocation

**TBD**.

The issued X.509 Certificates are related to the Federation Key attested within a
Trust Chain, their expiration time MUST be equal to the expiration of the Trust
Chain.

When a Federation Key is removed from the Requestor Entity Configuration
the X.509 Certificate related to it
SHOULD be revoked by its Credential Issuer, if not expired.

> A Requestor SHOULD request the revocation of its X.509 Certificate when the related
> when the related cryptographic material is revoked and published in the Federation Historical Key
> Registry.

Reviewer: ```suggestion
A Requestor SHOULD request the revocation of its X.509 Certificate when the related
cryptographic material is revoked and published in the Federation Historical Key
Registry.
```
editorial, duplicate words
